service/controller/chart/v1/resource/tillermigration/create.go

```diff
@@ -0,0 +1,100 @@
+package tillermigration
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/giantswarm/apiextensions/pkg/apis/application/v1alpha1"
+	"github.com/giantswarm/microerror"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/giantswarm/chart-operator/service/controller/chart/v1/key"
+)
+
+// EnsureCreated ensures Tiller is installed and the latest version.
+func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
+	cr, err := key.ToCustomResource(obj)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	// Resource is used to remove tiller pod in tenant clusters.
+	// So for other charts we can skip this step.
+	if key.ReleaseName(cr) != key.ChartOperatorName {
+		r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("no need to delete a tiller for %#q", key.ReleaseName(cr)))
+		r.logger.LogCtx(ctx, "level", "debug", "message", "canceling resource")
+		return nil
+	}
+
+	charts, err := r.g8sClient.ApplicationV1alpha1().Charts("giantswarm").List(metav1.ListOptions{})
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	notStarted := []string{}
+	inProgress := []string{}
+	for _, chart := range charts.Items {
+		foundConfigMap, err := r.findHelmV2ConfigMaps(ctx, chart)
+		if err != nil {
+			return microerror.Mask(err)
+		}
+
+		foundSecret, err := r.findHelmV3Secrets(ctx, chart)
+		if err != nil {
+			return microerror.Mask(err)
+		}
```

Comment:
How is the differentiation between v2 and v3 made? The code doesn't tell that story to me.

Reply:
V2 has release information as `configmap` resource in tiller namespace but V3 have these as `secret` (you can opt out to configmap with options) in release namespace. 
Having information on comments in both methods would be better?

Reply:
Comments would help to understand this but as this resource will be removed once the migration is sorted I do not mind.

Reply:
Then I prefer to delete the comments then.